tests/test_multidict.py

```diff
@@ -387,6 +387,17 @@ def test_and2(self, cls: Type[MutableMultiMapping[str]]) -> None:
 
         assert {"key"} == {"key", "key2"} & d.keys()
 
+    def test_and_not_implemented(self, cls: Type[MutableMultiMapping[str]]) -> None:
+        d = cls([("key", "value1")])
+
+        with pytest.raises(TypeError):
```

Comment:
It's usually important to use the `match=` regexp in the `pytest.raises()` and similar checks. Could you add it?

Reply:
OTOH, it might be cleaner to make a class with `__rand__`, and use its instance instead of `1`. You could track that it was called.

Reply:
This might not even need a literal class definition but a mock with that method and a check for how many times it was called + args.

Reply:
can you explain more about that?

Reply:
If the left operand's `__and__` returns `NotImplemented`, then Python would attempts calling `__rand__` from the right operand. So a properly crafted object type may have `__rand__(self, other)` defined that doesn't produce an exception. It could return a sentinel value for which we could check, for example.

Reply:
Here's an example without mocks: https://github.com/aio-libs/multidict/pull/936/files#r1473671524.

Reply:
Okay, now i got you, i will try to do so.